server/index-node/src/resolver.rs

```
@@ -470,17 +469,133 @@ where
+                    "Failed to query subgraph deployments";
+                    "subgraph" => subgraph_name,
+                    "errors" => format!("{:?}", result.errors)
+                );
+                return Ok(q::Value::Null);
+            }
+        };
+
+        let subgraphs = match data
+            .get_optional::<q::Value>("subgraphs")
+            .expect("invalid subgraphs")
+        {
+            Some(subgraphs) => subgraphs,
+            None => return Ok(q::Value::Null),
+        };
+
+        let mut subgraphs = subgraphs
+            .get_values::<q::Value>()
+            .expect("invalid subgraph values");
+
+        let subgraph = if let Some(subgraph) = subgraphs.pop() {
+            subgraph
+        } else {
+            return Ok(q::Value::Null);
+        };
+
+        let current = if let Some(current) = subgraph
+            .get_optional::<q::Value>("currentVersion")
+            .expect("invalid subgraph current version")
+        {
+            current
+        } else {
+            return Ok(q::Value::Null);
+        };
+
+        let deployment_id = current
+            .get_required::<q::Value>("deployment")
+            .expect("missing subgraph deployment")
+            .get_required::<String>("id")
+            .expect("invalid deployment id");
+
+        let block_number = ethereum_block_number.try_into().unwrap();
+        let deployment_id = SubgraphDeploymentId::new(deployment_id).unwrap();
+
+        let poi_fut = self
+            .store
+            .get_proof_of_indexing(&deployment_id, block_number);
+        let poi = match futures::executor::block_on(poi_fut) {
+            Ok(Some(poi)) => q::Value::String(poi.0),
+            Ok(None) => q::Value::Null,
+            Err(e) => {
+                error!(
+                    self.logger,
+                    "Failed to query proof of indexing";
+                    "subgraph" => subgraph_name,
+                    "error" => format!("{:?}", e)
+                );
```

Wouldn't we want to return the error to the client?

---

I would have thought so, but this is the pattern that other resolvers in the file are following.